src/backend/distributed/master/worker_node_manager.c

```diff
@@ -91,49 +74,7 @@ WorkerGetRandomCandidateNode(List *currentNodeList)
 		return NULL;
 	}
 
-	/* if current node list is empty, randomly pick one node and return */
-	if (currentNodeCount == 0)
-	{
-		workerNode = FindRandomNodeNotInList(WorkerNodesHash, NIL);
-		return workerNode;
-	}
-
-	/*
-	 * If the current list has an odd number of nodes (1, 3, 5, etc), we want to
-	 * place the shard on a different rack than the first node's rack.
-	 * Otherwise, we want to place the shard on the same rack as the first node.
-	 */
-	if (OddNumber(currentNodeCount))
-	{
-		wantSameRack = false;
-	}
-	else
-	{
-		wantSameRack = true;
```

Comment:
I have added the rack-related codes. I didn't make it nullable and marked the default rack as "default" :).